compiler/tests-common-jvm6/tests/org/jetbrains/kotlin/test/clientserver/TestProcessServer.kt

```diff
@@ -28,22 +28,37 @@ import java.util.concurrent.Executors
 import java.util.concurrent.ScheduledFuture
 import java.util.concurrent.TimeUnit
 
-fun getGeneratedClass(classLoader: ClassLoader, className: String): Class<*> {
+private fun ClassLoader.loadClassOrNull(name: String): Class<*>? =
     try {
-        return classLoader.loadClass(className)
+        loadClass(name)
+    } catch (e: ClassNotFoundException) {
+        null
     }
-    catch (e: ClassNotFoundException) {
-        error("No class file was generated for: " + className)
-    }
-}
 
-fun getBoxMethodOrNull(aClass: Class<*>): Method? {
+private fun Class<*>.getMethodOrNull(name: String, vararg parameterTypes: Class<*>): Method? =
     try {
-        return aClass.getMethod("box")
+        getMethod(name, *parameterTypes)
+    } catch (e: NoSuchMethodException) {
+        null
     }
-    catch (e: NoSuchMethodException) {
-        return null
+
+fun getGeneratedClass(classLoader: ClassLoader, className: String): Class<*> =
+    classLoader.loadClassOrNull(className) ?: error("No class file was generated for: $className")
+
+fun getBoxMethodOrNull(aClass: Class<*>): Method? =
+    aClass.getMethodOrNull("box")
+        ?: aClass.classLoader.loadClassOrNull("kotlin.coroutines.Continuation")?.let { aClass.getMethodOrNull("box", it) }
+        ?: aClass.classLoader.loadClassOrNull("kotlin.coroutines.experimental.Continuation")?.let { aClass.getMethodOrNull("box", it) }
+
+fun runBoxMethod(method: Method): String? {
+    if (method.parameterCount == 0) {
```

Comment:
@pyos There is also a problem with test execution on JDK6:
```
java.lang.NoSuchMethodError: java.lang.reflect.Method.getParameterCount()I
  at org.jetbrains.kotlin.test.clientserver.TestProcessServerKt.runBoxMethod(TestProcessServer.kt:54)
  at org.jetbrains.kotlin.test.clientserver.ServerTest.executeTest(TestProcessServer.kt:167)
```

Reply:
It's `compiler:tests-different-jdk:codegenTarget6Jvm6Test` task

Reply:
>public int getParameterCount()
>Since: 1.8

Huh. I guess it can be `parameterTypes.size` then.

Reply:
Done.